### PR TITLE
Migration of awesome_cossim_topn unit tests from wbaa name_matching p…

### PR DESCRIPTION
**Pull request by mbaak**
Hi Zhe, Ahmet,

At wbaa we're upgrading the entity-matching package. (It's used in multiple places now, with different purposes.) I noticed that the original awesome_cossim_topn unit tests were never migrated to this package. Here they are. I also updated awesome_cossim_topn() slightly to make sure that they pass the unit tests. Plus, I fixed flake8 errors.

Cheers, -Max

### CODE REVIEW

**Comment by mbaak** [comment]
If ok, can you please upload a new version on pypi? Thanks!

File: sparse_dot_topn/util.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+"""
+Helper functions for name matching
+"""
+import numpy as np
+
+
+def get_n_top_sparse(mat, n_top=10):
```
Comment (aerdem4) [inline sparse_dot_topn/util.py]: This seems to be only needed for the unit tests. Then it is better to have it under tests folder.

File: sparse_dot_topn/awesome_cossim_topn.py [hunk 2]
```diff
@@ -39,12 +39,21 @@ def awesome_cossim_topn(A, B, ntop, lower_bound=0, use_threads=False, n_jobs=1):
...
         err_str = 'A matrix multiplication will be operated. A.shape[1] must be equal to B.shape[0]!'
-        raise ValueError(err_str)        
+        raise ValueError(err_str)
 
     idx_dtype = np.int32
 
     nnz_max = M*ntop
 
+    # basic check. if A or B are all zeros matrix, return all zero matrix directly
+    if len(A.indices) == 0 or len(A.data) == 0 or len(A.indptr) == 0 or \
```
Comment (aerdem4) [inline sparse_dot_topn/awesome_cossim_topn.py]: Since they are csr matrix, if len(A.indices) is zero, rest will be zero too.

**Review by aerdem4 — COMMENTED** [review]
Thank you. I have added 2 comments. @ymwdalex maybe we can get Travis CI to run the unit tests automatically?

File: sparse_dot_topn/util.py (hunk 1 above)
Comment (mbaak) [reply sparse_dot_topn/util.py]: Moved.

File: sparse_dot_topn/awesome_cossim_topn.py (hunk 2 above)
Comment (mbaak) [reply sparse_dot_topn/awesome_cossim_topn.py]: Done.

Comment (mbaak) [reply sparse_dot_topn/awesome_cossim_topn.py]: Fixed. I also squashed the commits.

**Comment by mbaak** [comment]
The unit test of get_n_top_sparse() I'll keep in the original package (if still needed in fact.)

**Comment by ymwdalex** [comment]
The owner of this repository is ing-bank. In order to setup Travis CI, ing-bank github account need to give the permission. I have requested the permission, but don't know if they will approve it.
@aerdem4 @mbaak

**Comment by ymwdalex** [comment]
@mbaak could you please also update CHANGES.md file?

File: example/comparison.py [hunk 3]
```diff
@@ -5,9 +5,7 @@
 from __future__ import print_function
 import timeit
 import numpy as np
-from scipy.sparse import csr_matrix
 from scipy.sparse import rand
-from sparse_dot_topn import awesome_cossim_topn
```
Comment (aerdem4) [inline example/comparison.py:10]: probably you have removed it because flake8 complained? It is actually used within timeit as a string.

**Review by aerdem4 — COMMENTED** [review]
Added one more comment. The rest LGTM.

File: example/comparison.py (hunk 3 above)
Comment (mbaak) [reply example/comparison.py:10]: right ... fixed.

**Comment by mbaak** [comment]
I have updated the CHANGES.md file.

**Comment by mbaak** [comment]
Btw, let me know if you need help with the request for Travis CI.

**Comment by ymwdalex** [comment]
Looks good for me. Merged this PR

**Comment by ymwdalex** [comment]
@mbaak also package it and make it available on PyPi with version 0.2.9

**Comment by mbaak** [comment]
Great, thanks!

**Comment by ymwdalex** [comment]
@mbaak could you please push the ing-bank github administor to setup Travis.CI for this repository? Thanks!

Best regards,
Zhe

**Pull request merged**